src/components/GlobalNavbar/LoggedInNav.tsx

```diff
@@ -17,7 +17,13 @@ function LoggedInNav() {
   const { logOut } = useAuthActions();
   const user = useAuthenticatedUser() as User;
 
-  const walletAddress = user.addresses[0];
+  let userAddresses = user.addresses || [];
+
+  if (!userAddresses[0]) {
+    console.error('User is missing addresses');
+  }
+
+  const walletAddress = userAddresses[0] || '';
   const truncatedAddress = useMemo(() => {
     return truncate(walletAddress);
   }, [walletAddress]);
```

Comment:
i think we should just have this in a hook and have it throw an error; this will allow the Suspense boundary to catch the error, and we don't have to do `if X throw error` within the component itself

e.g.
```js
export default function useAuthenticatedUserAddress() {
  const user = userAuthenticatedUser()
  if (!user) {
    throw new Error('Authenticated user not found')
  }
  const address = user.addresses?.[0]
  if (!addresses) {
    throw new Error('No addresses found on user')
  }
  return address
}

...

function LoggedInNav() {
  ...
  const address = useAuthenticatedUserAddress()
  ...
}
```

Reply:
brilliant

Reply:
It renders the error via the Error Boundary momentarily then unmounts/shows the dev error message, is that expected for local? Console errors indicate the error wasn't caught which is odd `index.js:81 Uncaught Error: No address found on user`
![Screen Shot 2021-08-16 at 12 04 17 PM](https://user-images.githubusercontent.com/80802871/129506128-6a5865d5-7032-43d2-8753-9c3ec92771bc.png)